File: bench/fixtures/builds/typescript/make.ts
// Routines that make a Request, one per shape.
//
// `new X()` shape from .corpus/nestjs-nest/packages/core, which constructs its
// exception classes directly throughout.

import { make } from "./factory";
import { Request } from "./model";

/** PLAINLY TRUE: the construction is written here. */
export function build(): Request {
  return new Request("");
}

/** TRUE BUT HIDDEN: a factory in another file does the making. */
export function buildViaFactory(): Request {
  return make();
}

/** FALSE AND UNPROVABLE: this makes no Request. */
export function unrelated(): number {
  return 0;
}

File: bench/fixtures/builds/rust/src/make.rs
//! Routines that make a Request, one per shape.
//!
//! Struct-literal shape from .corpus/ripgrep/crates/globset/src/glob.rs:53 —
//! `MatchStrategy::BasenameLiteral(lit)` built inline in a constructor.

use crate::factory::make;
use crate::model::Request;

/// PLAINLY TRUE: the construction is written here.
pub fn build() -> Request {
    Request { path: String::new() }
}

/// TRUE BUT HIDDEN: this hands you a Request and never writes one, because a
/// factory in another file does the making.
pub fn build_via_factory() -> Request {
    make()
}

/// FALSE AND UNPROVABLE: this makes no Request, and SKILL.md says not finding a
/// construction proves nothing.
pub fn unrelated() -> usize {
    0
}
